src/Stratis.Bitcoin.Features.Consensus/PowConsensusValidator.cs

```diff
@@ -492,37 +492,22 @@ private long CountWitnessSignatureOperation(Script scriptSig, Script scriptPubKe
                 return 0;
 
             WitProgramParameters witParams = PayToWitTemplate.Instance.ExtractScriptPubKeyParameters2(scriptPubKey);
-            if (witParams != null)
-                return this.WitnessSigOps(witParams, witness, flags);
-
-            if (scriptPubKey.IsPayToScriptHash && scriptSig.IsPushOnly)
-            {
-                byte[] data = scriptSig.ToOps().Select(o => o.PushData).LastOrDefault() ?? new byte[0];
-                Script subScript = Script.FromBytesUnsafe(data);
```

Comment:
`subScript` is never used so I would suggest we go to the bitcoin core client and see how this is done there, it might be Nicolas forgot something

Reply:
Looks like he just found a way to do the same without those lines. 

This is BTC core code (interpreter.cpp:1567) : 
```
std::vector<unsigned char> data;
        while (pc < scriptSig.end()) {
            opcodetype opcode;
            scriptSig.GetOp(pc, opcode, data);
        }
        CScript subscript(data.begin(), data.end());
        if (subscript.IsWitnessProgram(witnessversion, witnessprogram)) {
            return WitnessSigOps(witnessversion, witnessprogram, witness ? *witness : witnessEmpty, flags);
        }
```

So here subScript is needed  to perform a check. In our code we use witParams for that check. 

So I think this was intentional.